sdk/identity/Azure.Identity/src/ClientSecretCredential.cs

```diff
@@ -51,6 +53,18 @@ public ClientSecretCredential(string tenantId, string clientId, string clientSec
         {
         }
 
+        /// <summary>
+        /// Creates an instance of the ClientSecretCredential with the details needed to authenticate against Azure Active Directory with a client secret.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="clientSecret">A client secret that was generated for the App Registration used to authenticate the client.</param>
+        /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
+        public ClientSecretCredential(string tenantId, string clientId, string clientSecret, ClientSecretCredentialOptions options)
+            : this(tenantId, clientId, clientSecret, (TokenCredentialOptions)options)
```

Comment:
Why this cast `(TokenCredentialOptions)options` is required?

Reply:
I added it because I wanted to be explicit about calling the overload which take `TokenCredential`. Perhaps it's not needed if this(*) will never recurse.